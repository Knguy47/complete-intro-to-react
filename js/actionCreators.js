import axios from 'axios';
import { SET_SEARCH_TERM, ADD_API_DATA } from './actions';

export function setSearchTerm(searchTerm) {
  return { type: SET_SEARCH_TERM, payload: searchTerm, error: null };
}

export function clearSearchTerm(searchTerm) {
  return { type: SET_SEARCH_TERM, payload: searchTerm };
}

export function addAPIData(apiData) {
  return { type: ADD_API_DATA, payload: apiData};
}

export function getAPIDetails(imdbID) {
  console.log(imdbID); // eslint-disable-line no-console
  return (dispatch) => {
    axios
      .get(`http://localhost:3000/${imdbID}`)
      .then(response => {
        console.log(response.data) // eslint-disable-line no-console
        dispatch(addAPIData(response.data));
      })
      .catch(err => {
        console.error('axios error', err); // eslint-disable-line no-console
      });
  }
}