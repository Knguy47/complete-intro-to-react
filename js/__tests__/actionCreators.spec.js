import moxios from 'moxios';
import { setSearchTerm, addAPIData, getAPIDetails } from '../actionCreators';

const westWorld = {
  "title": "Westworld",
  "year": "2016â€“",
  "description": "Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.",
  "poster": "ww.jpg",
  "imdbID": "tt0475784",
  "trailer": "eX3u0IlBBO4",
  "rating": "8.6"
};

test('setSearchTerm', () => {
  expect(setSearchTerm('Orange')).toMatchSnapshot();
});

test('addAPIData', () => {
  expect(addAPIData(westWorld)).toMatchSnapshot();
});

test('getAPIDetails', (done) => {
  const dispatchMock = jest.fn();
  
  moxios.withMock(() => {
    getAPIDetails(westWorld.imdbID)(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: westWorld
        })
        .then(() => {
          expect(request.url).toEqual(`http://localhost:3000/${westWorld.imdbID}`);
          expect(dispatchMock).toBeCalledWith(addAPIData(westWorld));
          done();
        });
    });
  });
});