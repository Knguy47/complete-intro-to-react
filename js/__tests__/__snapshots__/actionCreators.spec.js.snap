// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.",
    "imdbID": "tt0475784",
    "poster": "ww.jpg",
    "rating": "8.6",
    "title": "Westworld",
    "trailer": "eX3u0IlBBO4",
    "year": "2016â€“",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "error": null,
  "payload": "Orange",
  "type": "SET_SEARCH_TERM",
}
`;
